packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/MediaLibraryUtils.java

```diff
@@ -213,25 +225,66 @@ static Integer convertMediaType(String mediaType) throws IllegalArgumentExceptio
     return MEDIA_TYPES.get(mediaType);
   }
 
-  static int[] getSizeFromCursor(Cursor cursor, int mediaType, int localUriIndex){
+  static int[] getSizeFromCursor(ContentResolver contentResolver, ExifInterface exifInterface, Cursor cursor, int mediaType, int localUriIndex) {
+    final String uri = cursor.getString(localUriIndex);
+
+    if (mediaType == Files.FileColumns.MEDIA_TYPE_VIDEO) {
+      Uri photoUri = Uri.parse("file://" + uri);
+      try {
+        AssetFileDescriptor photoDescriptor = contentResolver.openAssetFileDescriptor(photoUri, "r");
+        MediaMetadataRetriever retriever = new MediaMetadataRetriever();
```

Comment:
These variables can be wrap into the try-with-resources structure. 
So, it should look like this:
```java
try (AssetFileDescriptor photoDescriptor = contentResolver.openAssetFileDescriptor(photoUri, "r");
     MediaMetadataRetriever retriever = new MediaMetadataRetriever()) {
...
}
```

It'll make this code more readable.

Reply:
Unfortunately `MediaMetadataRetriever` is not `AutoCloseable`, but `AssetFileDescriptor` is. I'll move the latter to use try-with-resources and leave the former as try-finally

Reply:
Ohh, you are right. It's `Autocloseable` in the newer Android version.